src/journal/journalctl.c

```diff
@@ -2397,8 +2397,10 @@ int main(int argc, char *argv[]) {
                         goto finish;
                 }
                 if (r < 0) {
+                        /* inotify resources exhausted, can't get wd */
                         log_error_errno(r, "Failed to get journal fd: %m");
-                        goto finish;
+                        log_error("Insufficent watch descriptors available. Reverting to -n");
```

Comment:
Hmm, this is too simple, there might be other reasons why sd_journal_get_fd() might fail... This needs to check explicitly the error condition, i.e. the error code inotify_add_watch() generates in this case. Also, please terminate the sentence with a full stop.  And this should really be a log_warning(), not a log_error(), as we continue operation to some level after this happened.